### PR TITLE
net: wifi: shell: add enterprise support for sap

### PR DESCRIPTION
**Pull request by Rex-Chen-NXP**
Add EAP-TLS, EAP-PEAP-MSCHAPv2, EAP-PEAP-GTC, EAP-TTLS-MSCHAPv2, EAP-PEAP-TLS, EAP-TLS-SHA256 enterprise wpa2 and wpa3 suiteb support for sap.

### CODE REVIEW

**Comment by Rex-Chen-NXP** [comment]
Hi @krish2718 @jukkar 
Could you help review this PR, also the below one.
https://github.com/zephyrproject-rtos/hostap/pull/63

Thanks!

**Comment by krish2718** [comment]
> Hi @krish2718 @jukkar Could you help review this PR, also the below one. [zephyrproject-rtos/hostap#63](https://github.com/zephyrproject-rtos/hostap/pull/63)
> 
> Thanks!

I guess this would be for 4.1.0 now? So, may be we can prioritize other PRs

**Comment by Rex-Chen-NXP** [comment]
> > Hi @krish2718 @jukkar Could you help review this PR, also the below one. [zephyrproject-rtos/hostap#63](https://github.com/zephyrproject-rtos/hostap/pull/63)
> > Thanks!
> 
> I guess this would be for 4.1.0 now? So, may be we can prioritize other PRs

Yes, of course, follow your review priority

**Comment by Rex-Chen-NXP** [comment]
Hi @krish2718 @jukkar 
Could you give high priority to review this patch after 2024/11/15(code free).
Thanks!

File: include/zephyr/net/wifi.h [hunk 1]
```diff
@@ -237,6 +237,8 @@ const char *wifi_band_txt(enum wifi_frequency_bands band);
 #define WIFI_ENT_IDENTITY_MAX_LEN 64
 /** Max enterprise password length */
 #define WIFI_ENT_PSWD_MAX_LEN 128
+/** Max enterprise identity users */
+#define WIFI_ENT_IDENTITY_MAX_USERS 8
```
Comment (krish2718) [inline include/zephyr/net/wifi.h:241]: This doesn't belong in this file, this is only for generic/protocol related stuff.

Comment (krish2718) [reply include/zephyr/net/wifi.h:241]: and this should be part of Kconfig.

File: modules/hostap/CMakeLists.txt [hunk 2]
```diff
@@ -579,6 +579,72 @@ zephyr_library_sources_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_DPP
...
+	${HOSTAP_SRC_BASE}/eap_server/eap_server_mschapv2.c
+)
+
+zephyr_library_compile_definitions_ifdef(CONFIG_EAP_SERVER_MSCHAPV2
+        EAP_SERVER_MSCHAPV2
+)
+
+zephyr_library_sources_ifdef(CONFIG_EAP_SERVER_PEAP
+	${HOSTAP_SRC_BASE}/eap_server/eap_server_peap.c
+)
```
Comment (krish2718) [inline modules/hostap/CMakeLists.txt:626]: alignment is mixed

File: modules/hostap/Kconfig [hunk 3]
```diff
@@ -265,6 +265,30 @@ config WIFI_NM_HOSTAPD_AP
 	bool "FullAP mode support based on Hostapd"
 	depends on !WIFI_NM_WPA_SUPPLICANT_INF_MON
 
+config WIFI_NM_HOSTAPD_CRYPTO_ENTERPRISE
+	bool "Hostapd crypto enterprise support"
+
+config EAP_SERVER_TLS
```
Comment (krish2718) [inline modules/hostap/Kconfig:271]: Should we also add `EAP_SERVER_ALL` like we did for STA?

File: modules/hostap/src/supp_api.c [hunk 4]
```diff
@@ -550,6 +553,253 @@ static int is_eap_valid_security(int security)
...
+		wpa_printf(MSG_ERROR, "EAP user allocation failed");
+		goto failed;
+	}
+
+	user->force_version = -1;
+	if (params->eap_ver >= 0) {
+		user->force_version = params->eap_ver;
+	}
+
+	if (hapd_get_eap_config(params, &eap_cfg))
```
Comment (krish2718) [inline modules/hostap/src/supp_api.c:618]: missing braces

File: modules/hostap/src/supp_api.c [hunk 5]
```diff
@@ -550,6 +553,253 @@ static int is_eap_valid_security(int security)
...
+		tail->next = user;
+		tail       = user;
+	}
+
+	*pnew_user = new_user;
+
+	return tail;
+
+failed:
+	if (user)
```
Comment (jukkar) [inline modules/hostap/src/supp_api.c:636]: Braces missing

File: modules/hostap/src/supp_api.c [hunk 6]
```diff
@@ -550,6 +553,253 @@ static int is_eap_valid_security(int security)
...
+static int hostapd_process_eap_user(struct wifi_connect_req_params *params,
+	struct hostapd_bss_config *conf)
+{
+	struct hostapd_eap_user *user = NULL, *tail = NULL, *user_list = NULL;
+	uint8_t i, nusers = params->nusers;
+	const char *identity, *password;
+	struct wifi_eap_config eap_cfg;
+	int ret = 0;
+
+	if (hapd_get_eap_config(params, &eap_cfg))
```
Comment (jukkar) [inline modules/hostap/src/supp_api.c:651]: braces missing

File: modules/hostap/src/supp_api.c [hunk 7]
```diff
@@ -550,6 +553,253 @@ static int is_eap_valid_security(int security)
...
+		goto failed;
+
+	if (eap_cfg.phase2 != NULL)
```
Comment (jukkar) [inline modules/hostap/src/supp_api.c:654]: here too

File: modules/hostap/src/supp_api.c [hunk 8]
```diff
@@ -550,6 +553,253 @@ static int is_eap_valid_security(int security)
...
+			tail = user_list = user;
+		} else {
+			tail->next = user;
+			tail       = user;
+		}
+
+		continue;
+
+failed:
+		if (user)
```
Comment (jukkar) [inline modules/hostap/src/supp_api.c:717]: ditto

File: modules/hostap/src/supp_api.c [hunk 9]
```diff
@@ -550,6 +553,253 @@ static int is_eap_valid_security(int security)
...
+		hostapd_config_free_eap_users(conf->eap_user);
+		conf->eap_user = user_list;
+	} else {
+		hostapd_config_free_eap_users(user_list);
+	}
+
+	return ret;
+}
+
+int hapd_process_enterprise_config(struct hostapd_iface *iface,
```
Comment (jukkar) [inline modules/hostap/src/supp_api.c:733]: this function has `if` checks that are missing braces, please fix all of them

File: include/zephyr/net/wifi.h (hunk 1 above)
Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi.h]: Updated as comment

File: modules/hostap/CMakeLists.txt (hunk 2 above)
Comment (Rex-Chen-NXP) [reply modules/hostap/CMakeLists.txt]: Updated as comment

File: modules/hostap/Kconfig (hunk 3 above)
Comment (Rex-Chen-NXP) [reply modules/hostap/Kconfig:271]: Yes. of course, added  EAP_SERVER_ALL

File: modules/hostap/src/supp_api.c (hunk 4 above)
Comment (Rex-Chen-NXP) [reply modules/hostap/src/supp_api.c]: Fixed as comment

File: modules/hostap/src/supp_api.c (hunk 5 above)
Comment (Rex-Chen-NXP) [reply modules/hostap/src/supp_api.c]: Fixed as comment

File: modules/hostap/src/supp_api.c (hunk 7 above)
Comment (Rex-Chen-NXP) [reply modules/hostap/src/supp_api.c]: Fixed as comment

File: modules/hostap/src/supp_api.c (hunk 8 above)
Comment (Rex-Chen-NXP) [reply modules/hostap/src/supp_api.c]: Fixed as comment

File: modules/hostap/src/supp_api.c (hunk 9 above)
Comment (Rex-Chen-NXP) [reply modules/hostap/src/supp_api.c:739]: Added all missing braces as comment

**Comment by Rex-Chen-NXP** [comment]
Hi @krish2718  @jukkar 
Could you help again? And when this PR can merged if approved by reviewers?

**Review by krish2718 — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Comment by jukkar** [comment]
@Rex-Chen-NXP please update the manifest as the hostap PR is now merged.

**Comment by Rex-Chen-NXP** [comment]
> @Rex-Chen-NXP please update the manifest as the hostap PR is now merged.

Updated.

**Comment by Rex-Chen-NXP** [comment]
Hi @krish2718  @jukkar 
Could you help review again and merge PR?
Thanks.

**Review by krish2718 — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**